server/protocols/webadmin/webadmin-data/src/main/java/org/apache/james/webadmin/routes/GroupsRoutes.java

```diff
@@ -121,6 +122,50 @@ public HaltException createGroupWithDummyUser(Request request, Response response
         return halt(HttpStatus.NO_CONTENT_204);
     }
 
+    public static class GroupStatusInfo {
+        // Fields
+        public String address;
+        public String status;
+        public String reason;
+
+        // Constructor
+        public GroupStatusInfo(String address, String status, String reason) {
+            this.address = address;
+            this.status = status;
+            this.reason = reason;
+        }
+    }
+
+    public String createMultipleGroupWithDummyUser(Request request, Response response) throws JsonProcessingException {
+
+        String jsonString = request.body();
+        List<String> groups = objectMapper.readValue(jsonString, new TypeReference<List<String>>() {});
+
+        GroupStatusInfo[] result = new GroupStatusInfo[groups.size()];
+
+        for (int i = 0; i < groups.size(); i++) {
+            String group = groups.get(i);
+            MailAddress groupAddress;
+            try {
+                groupAddress = new MailAddress(group);
+                Domain domain = groupAddress.getDomain();
+                MappingSource source = MappingSource.fromUser(Username.fromLocalPartWithDomain(groupAddress.getLocalPart(), domain));
+                addGroupMember(source, dummyUser);
+                result[i] = new GroupStatusInfo(group, "success","");
+            } catch (Exception e) {
+                if (e.toString().equals("spark.HaltException")) {
+                    result[i] = new GroupStatusInfo(group, "failed", "Source domain is not managed by the domainList.Domain does not exist");
```

Comment:
Rewrite this error message: The source domain is not recognized or does not exist in the domain list.